models/users.js

```diff
@@ -317,6 +317,16 @@ const fetchUser = async ({ userId = null, username = null, githubUsername = null
         userData = doc.data();
       });
     }
+
+    if (userData && userData.disabled_roles !== undefined) {
+      const updatedRoles = { ...userData.roles };
```

Comment:
So here you are spreading a object inside an object, is it not negating it self?

will it not be the same as `updatedRoles = userData.roles`?

Reply:
Wanted to have a deep copy of the `userData.roles` object and mutate it independently without mending with the roles, but yes we can do simple assignment as well.